cilium/CFP-28986-mutual-auth-updates.md

```diff
@@ -0,0 +1,256 @@
+
+This CFP lays out a proposal to mitigate this risk in two ways:
+* Require the use of either encryption (recommended because it prevents man-in-the-middle attacks) or encapsulation (which prevents cache manipulation but not MITM). In both of these cases, the Cilium identity is included in every packet, and so the ipcache is not used to impute identity to packets.
+* Add an additional mutual authentication mode that will use the connection tracking ("conntrack") table in the conntrack BPF map to store information about what _connections_ need authentication, instead of what _identity-identity-node pairings_ need authentication.
+
+The first idea makes cache manipulation irrelevant, because the inbound packets include the security identity in a header.
+
+However, in the existing mode, mutual authentication handshakes are only initiated when there is a change in either endpoint state (that is, a new node to node connection from a new pod), or when a certificate is rotated. So, in the event that an attacker _did_ manage to compromise mutual authentication mechanisms somehow, the window is, at maximum, the certificate rotation period, which defaults to 30 minutes.
+
+Adding a connection-based mutual authentication mode significantly reduces the timescale that an attack is exploitable for - since the mutual auth handshake is performed on each new connection, and connections _generally_ last much less that 30 minutes.
+
+It also makes performing the exploit harder, since the full 5-tuple (source IP, source port, destination IP, destination port, protocol) must match, rather than just source and destination IP, _and_ must also migrate the actual connection from one application to another. Still possible, but orders of magnitude harder.
+
+## Goals
+
+* Document and enforce the new requirement to have some form of encapsulation (either encryption, VXLAN, or similar) enabled when mutual auth is enabled.
+* Add support for mutual authentication on a per-connection basis in addition to the existing per-identity-pair mutual authentication.
+
+## Non-Goals
+
+* A mutual authentication design that works with Cilium Cluster Mesh. This is left for future work.
+
+## Proposal
+
+### Overview
+
+This CFP proposes two changes: enforcing every packet contains an identity using either encryption or encapsulation and adding a per-connection mutual auth mode. These changes are relatively orthogonal and could be implemented as two separate changes.
+
+#### Requiring encryption or encapsulation resolves cache manipulation
+
+This assertion depends on two things that may not be clear:
+
+- Wireguard encryption now (as of Cilium 1.15) also encapsulates packets into VXLAN as they transit the tunnel, adding a Cilium identity into the header (as in unencrypted VXLAN)
```

Comment:
IPSec as well?

Reply:
I'm not sure - the IPSec code has been going through signfiicant refactoring in the last few months, so I thought it was better to leave it out.